.github/workflows/main.yml

```diff
@@ -110,80 +104,109 @@ jobs:
           name: unit_results_${{ matrix.python-version }}-${{ steps.date.outputs.date }}.csv
           path: unit_results.csv
 
-  build:
-    name: build packages
+  integration:
+    name: integration test / python ${{ matrix.python-version }} / ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+        os: [ubuntu-latest]
+        include:
+          - python-version: 3.8
+            os: windows-latest
+          - python-version: 3.8
+            os: macos-latest
+
+    env:
+      TOXENV: integration
+      PYTEST_ADDOPTS: "-v --color=yes -n4 --csv integration_results.csv"
+      DBT_INVOCATION_ENV: github-actions
 
     steps:
       - name: Check out the repository
         uses: actions/checkout@v2
-        with:
-          persist-credentials: false
 
-      - name: Set up Python
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
-          python-version: 3.8
+          python-version: ${{ matrix.python-version }}
 
-      - name: Install python dependencies
-        run: |
-          pip install --user --upgrade pip
-          pip install --upgrade setuptools wheel twine check-wheel-contents
-          pip --version
+      - name: Set up postgres (linux)
+        if: runner.os == 'Linux'
+        uses: ./.github/actions/setup-postgres-linux
 
-      - name: Build distributions
-        run: ./scripts/build-dist.sh
+      - name: Set up postgres (macos)
+        if: runner.os == 'macOS'
+        uses: ./.github/actions/setup-postgres-macos
 
-      - name: Show distributions
-        run: ls -lh dist/
+      - name: Set up postgres (windows)
+        if: runner.os == 'Windows'
+        uses: ./.github/actions/setup-postgres-windows
 
-      - name: Check distribution descriptions
+      - name: Install python tools
         run: |
-          twine check dist/*
+          pip install --user --upgrade pip
+          pip --version
+          pip install tox
+          tox --version
 
-      - name: Check wheel contents
-        run: |
-          check-wheel-contents dist/*.whl --ignore W007,W008
+      - name: Run tests
+        run: tox
 
       - uses: actions/upload-artifact@v2
+        if: always()
         with:
-          name: dist
-          path: dist/
+          name: logs
```

Comment:
Should we follow a similar naming convention like we do for unit tests in line 104?

Reply:
Oh, I see what you mean. This will clobber the files on each permutation of matrix. I'll update!